sematic/api/endpoints/events.py

```diff
@@ -16,27 +19,66 @@
 
 logger = logging.getLogger(__name__)
 
+_sio_server = None
 
-@sematic_api.route("/api/v1/events/<namespace>/<event>", methods=["POST"])
+_ROUTE = "/api/v1/events/<namespace>/<event>"
+_STARLETTE_ROUTE = _ROUTE.replace("<", "{").replace(">", "}")
+
+def register_sio_server(sio_server):
+    global _sio_server
+    if _sio_server is not None:
+        raise RuntimeError("SocketIO Server already registered.")
+    _sio_server = sio_server
+
+
+@sematic_api.route(_ROUTE, methods=["POST"])
 @authenticate
-def events(user: Optional[User], namespace: str, event: str) -> flask.Response:
+def sync_events(user: Optional[User], namespace: str, event: str) -> flask.Response:
     """
     Sends out a socketio broadcast notification to all subscribed listeners (Resolvers,
     the Dashboard, etc.).
     """
     logger.info("Broadcasting: namespace=%s; event=%s", namespace, event)
     logger.debug("Broadcasting: json payload=%s", flask.request.json)
 
-    flask_socketio.emit(
+    _sio_server.emit(
         event,
         flask.request.json,
-        namespace="/{}".format(namespace),
-        broadcast=True,
+        namespace=f"/{namespace}",
     )
 
     return flask.jsonify({})
 
 
+async def async_events(request):
+    """
+    Sends out a socketio broadcast notification to all subscribed listeners (Resolvers,
+    the Dashboard, etc.).
+    """
```

Comment:
Need to add auth to this

Reply:
is this comment outdated?  what else needs to be done other than `@authenticate_starlette`?

Reply:
yup, comment is outdated. I have now added auth